### PR TITLE
adb: Update to version 28.0.3

### PR DESCRIPTION
**Pull request by TheCjw**
Update adb to 28.0.3 and fix autoupdate.

Just like sdk manager of android studio, we could use https://dl.google.com/android/repository/repository2-1.xml to check latest sdk tools.

### CODE REVIEW

File: bucket/adb.json [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
         "platform-tools\\adb.exe",
         "platform-tools\\dmtracedump.exe",
         "platform-tools\\etc1tool.exe",
         "platform-tools\\fastboot.exe",
         "platform-tools\\hprof-conv.exe"
     ],
-    "checkver": "<h4.*>([\\d.]+) \\(.*\\)</h4>",
+    "checkver": {
+        "url": "https://dl.google.com/android/repository/repository2-1.xml",
+        "re": "platform-tools_r(?<version>[\\d.]+)-windows.zip"
```
Comment (h404bi) [inline bucket/adb.json]: We can use XPath to query the version and hash.

Comment (TheCjw) [reply bucket/adb.json]: Woo, that's a nice feature, but wiki isn't update yet.

I'll try, wait a second.

Comment (TheCjw) [reply bucket/adb.json]: After some testing, I think XPath make things complicated.

Google puts version code in sperated nodes:

```xml
	<remotePackage path="platform-tools">
		<!--Generated from:ab bid:5475833 branch:aosp-sdk-release-->
		<type-details xsi:type="generic:genericDetailsType"/>
		<revision>
			<major>28</major>
			<minor>0</minor>
			<micro>3</micro>
		</revision>
```

To get full version string, we have to use `concat` function, the xpath looks terrible: 

```JavaScript
// Tested with Chrome
$x("concat(//remotePackage[@path='platform-tools']/revision/major, '.', //remotePackage[@path='platform-tools']/revision/minor, '.', //remotePackage[@path='platform-tools']/revision/micro)")
// Result: "28.0.3"
```

And the worst thing is, powershell did not supported `concat` function. So regular expression is the simplest way for this.

**Comment by TheCjw** [comment]
@Ash258 Thanks.

**Comment by Ash258** [comment]
Add license. It is in my snippet.

**Comment by TheCjw** [comment]
@Ash258 License added.

File: bucket/adb.json [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
-    "version": "28.0.2",
+    "version": "28.0.3",
     "homepage": "https://developer.android.com/studio/releases/platform-tools.html",
+    "license": "Apache-2.0",
     "description": "Android SDK platform-tools includes tools that interface with the Android platform, such as adb, fastboot, and systrace",
-    "url": "https://dl.google.com/android/repository/platform-tools_r28.0.2-windows.zip",
-    "hash": "6a721560633babebc74f9330d7184a23d74d18b835cc769f81bbf977575b3800",
+    "url": "https://dl.google.com/android/repository/platform-tools_r28.0.3-windows.zip",
+    "hash": "sha1:6d2db43c69a71bcd7b897b5386e065dfa713b340",
     "bin": [
```
Comment (Ash258) [inline bucket/adb.json]:
```suggestion
    "extract_dir": "platform-tools",
    "bin": [
```

Comment (TheCjw) [reply bucket/adb.json]: Suggestion applied.

File: bucket/adb.json [hunk 3]
```diff
@@ -1,18 +1,27 @@
...
+    "version": "28.0.3",
     "homepage": "https://developer.android.com/studio/releases/platform-tools.html",
+    "license": "Apache-2.0",
     "description": "Android SDK platform-tools includes tools that interface with the Android platform, such as adb, fastboot, and systrace",
-    "url": "https://dl.google.com/android/repository/platform-tools_r28.0.2-windows.zip",
-    "hash": "6a721560633babebc74f9330d7184a23d74d18b835cc769f81bbf977575b3800",
+    "url": "https://dl.google.com/android/repository/platform-tools_r28.0.3-windows.zip",
+    "hash": "sha1:6d2db43c69a71bcd7b897b5386e065dfa713b340",
+    "extract_dir": "platform-tools",
     "bin": [
```
Comment (Ash258) [inline bucket/adb.json]:
```suggestion
    "bin": [
        "adb.exe",
        "dmtracedump.exe",
        "etc1tool.exe",
        "fastboot.exe",
        "hprof-conv.exe"
    ],
```

**Pull request closed without merging**